tools/batch/batch_states/compile_notebooks.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# Shell script for submitting AWS Batch jobs to compile notebooks
+
+prnumber=$1
+runnumber=$2
+remote=$3
+refs=$4
+
+compile_notebook () {
+    local MDFILE=$1
+    DIR=$(dirname $MDFILE)
+    BASENAME=$(basename $MDFILE)
+    TARGETNAME=$(dirname $MDFILE)/${BASENAME%.md}.ipynb
+    LOGNAME=${BASENAME%.md}.stdout.log
+    JOBIDLOG=${BASENAME%.md}.jobid.log
+
+    echo Compiling $BASENAME ...
+
+    python3 tools/batch/submit-job.py --region us-east-1 \
+            --wait \
+            --timeout 3600 \
+            --saved-output /gluon-nlp/docs/examples \
+            --name GluonNLP-Docs-${refs}-${prnumber}-${runnumber} \
+            --save-path ${runnumber}/gluon-nlp/docs/examples \
+            --work-dir . \
+            --source-ref ${refs} \
+            --remote https://github.com/${remote} \
+            --command "python3 -m pip install --quiet nbformat notedown jupyter_client ipykernel && \
+                       python3 /gluon-nlp/docs/md2ipynb.py ${MDFILE}" 2>&1 | tee $LOGNAME >/dev/null
+
+    BATCH_EXIT_CODE=$?
+
+    head -100 $LOGNAME | grep -oP -m 1 'jobId: \K(.*)' > $JOBIDLOG
+
+    JOBID=$(cat "$JOBIDLOG")
+
+    python3 tools/batch/wait-job.py --job-id $JOBID
```

Comment:
this will make all the notebooks run sequentially on Batch, which defeats the purpose of using a job management system

instead, the script should submit all notebook jobs, collect the job ids, and then use a separate loop to wait on them.

Reply:
In fact, I was using `&` to parallelize for-loop, so that these jobs can be run simultaneously.

Reply:
ah ok, totally missed it on line 51. thanks for clarifying.